onboarding_auth/onboardingAuthData/src/main/java/com/alexafit/onboardingauthdata/repository/OnboardingAuthRepositoryImpl.kt

```diff
@@ -0,0 +1,34 @@
+package com.alexafit.onboardingauthdata.repository
+
+import com.alexafit.onboardingauthdata.model.remote.Login
+import com.alexafit.onboardingauthdata.model.remote.LoginResponse
+import com.alexafit.onboardingauthdata.model.remote.Register
+import com.alexafit.onboardingauthdata.model.remote.RegisterResponse
+import com.alexafit.onboardingauthdata.remote.TaskyApi
+
+class OnboardingAuthRepositoryImpl(
+    private val taskyApi: TaskyApi
+) : OnboardingAuthRepository {
+    override suspend fun registerUser(registerBody: Register): Result<RegisterResponse> {
+        val registerResponse = taskyApi.registerUser(register = registerBody)
+        return if (registerResponse.isSuccessful) {
+            Result.success(RegisterResponse("Success"))
+        } else {
+            Result.failure(Throwable(message = registerResponse.body()?.message))
+        }
+    }
+
+    override suspend fun loginUser(loginBody: Login): Result<LoginResponse> {
```

Comment:
LoginResponse its a data object

I'd ceeate a similar class for the domain model, and use that one by mapping the data stuff to the domain

This helps make it so presentation doesnt know data stuff